File: lib/bean/media_notification_info.dart
/*
* 作者 Ren
* 时间  2023/7/4 21:12
*/
class MediaNotificationInfo {
  String? title;
  String? subtitle;
  bool? isNextEnable;
  bool? isPreviousEnable;
  bool? isPlaying;
  int? duration;
  int? position;
  double? playSpeed;
  String? imagePath;
  String? placeHolderAssets;
  List<int>? imageData;

  MediaNotificationInfo({
    this.title,
    this.subtitle,
    this.isNextEnable,
    this.isPreviousEnable,
    this.isPlaying,
    this.duration,
    this.position,
    this.playSpeed,
    this.imagePath,
    this.placeHolderAssets,
    this.imageData,
  });

  Map toMap() {
    return {
      'title': title,
      'subtitle': subtitle,
      'isNextEnable': isNextEnable,
      'isPreviousEnable': isPreviousEnable,
      'isPlaying': isPlaying,
      'duration': duration,
      'position': position,
      'playSpeed': playSpeed,
      'imagePath': imagePath,
      'imageData': imageData
    };
  }
}

File: android/src/main/kotlin/com/nl/media/notification/bean/MediaNotificationInfo.kt
package com.nl.media.notification.bean

import com.nl.media.notification.config.NotificationConfig

class MediaNotificationInfo(
    var title: String? = null,
    var subtitle: String? = null,
    var hasNext: Boolean? = null,
    var hasPre: Boolean? = null,
    var isPlaying: Boolean? = null,
    var duration: Long? = null,
    var position: Long? = null,
    var playSpeed: Float? = null,
    var placeHolderAssets: String? = null,
    var imagePath: String? = null,
    var imageData: ByteArray? = null,
) {
    companion object {
        fun fromMap(map: Map<String, Any?>): MediaNotificationInfo {
            return MediaNotificationInfo(
                title = map["title"] as? String?,
                subtitle = map["subtitle"] as? String?,
                hasNext = map["hasNext"] as? Boolean?,
                hasPre = map["hasPre"] as? Boolean?,
                isPlaying = map["isPlaying"] as? Boolean?,
                duration = (map["duration"] as? Int?)?.toLong(),
                position = (map["position"] as? Int?)?.toLong(),
                playSpeed = map["playSpeed"] as? Float?,
                placeHolderAssets = map["placeHolderAssets"] as? String?,
                imagePath = map["imagePath"] as? String?,
                imageData = map["imageData"] as? ByteArray?,
            )
        }
    }
}